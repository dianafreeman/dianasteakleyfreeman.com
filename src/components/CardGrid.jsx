/* eslint-disable react/destructuring-assignment */
/* eslint-disable no-return-assign */
/* eslint-disable react/prop-types */
import { FaChevronLeft, FaChevronRight } from 'react-icons/fa'
import React from 'react'
import Slider from 'react-slick'
import styled from 'styled-components'
import tw from 'tailwind.macro'
import ProjectCard from './ProjectCard'
import 'slick-carousel/slick/slick.css'
import 'slick-carousel/slick/slick-theme.css'

const GridContainer = styled.div`
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
`

const GridRow = styled.div`
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
`

class CardGrid extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      projects: this.props.projects,
    }
  }
  // get the number of projects
  // chooose three per row

  render() {
    return (
      <GridContainer>
        <GridRow>
        {this.state.projects.map(project => (
          <ProjectCard
            title={project.title}
            bg={project.acf.project_color}
            key={`proj-card-${project.wordpress_id}`}
            img={project.featured_media.localFile.childImageSharp.resolutions.src}
          />
        ))}
        </GridRow>
      </GridContainer>
    )
  }
}
export default CardGrid
