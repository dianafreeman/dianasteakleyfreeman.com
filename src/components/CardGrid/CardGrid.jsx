/* eslint-disable react/require-default-props */

import React, { useState } from 'react';
import { useTrail, animated } from 'react-spring';
import Card from '../Card';

const items = ['item a', 'item b', 'item c', 'item d', 'item e', 'item f'];

const CardGrid = ({ props }) => {
  const [show, set] = useState(false);
  const config = { mass: 5, tension: 2000, friction: 200 };
  const trail = useTrail(items.length, {
    config,
    o: show ? 1 : 0,
    s_xyz: show ? [1, 1, 0.5] : [1, 1, -1],
    from: { x: 0, h: 0, o: 1 },
  });

  const AniCard = animated(Card);

  return (
    <div onClick={() => set(state => !state)}>
      {trail.map(({ s_xyz, o, ...rest }, index) => (
        <AniCard
          key={items[index]}
          styles={{
            opacity: o,
            transform: s_xyz.interpolate(
              (x, y, z) => `perspective(95vw) scale3d(${x},${y},${z}) translateZ(-100px)`
            ),
            ...rest,
          }}
          title={items[index]}
        />
      ))}
    </div>
  );
};

export default CardGrid;
