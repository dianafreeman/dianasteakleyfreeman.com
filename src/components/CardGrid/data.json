[
  {
    "node": {
      "id": "b114c8e7-3cb9-5685-b8cc-af0a9b3fd202",
      "frontmatter": {
        "slug": "eliza-theraupeutic-ai",
        "path": "/blog/eliza-theraupeutic-ai",
        "date": "2019-04-03"
      }
    }
  },
  {
    "node": {
      "id": "58dc1e9c-74d9-5f5c-ae81-0a1b562326f6",
      "frontmatter": {
        "slug": "user-misdirection",
        "path": "/blog/user-misdirection",
        "date": "2019-04-05"
      }
    }
  },
  {
    "node": {
      "id": "9a34bc76-90a6-50ed-b2e6-6212e7c2bd28",
      "frontmatter": {
        "slug": "digital-self-care",
        "path": "/blog/digital-self-care",
        "date": "2019-03-31"
      }
    }
  }
]
