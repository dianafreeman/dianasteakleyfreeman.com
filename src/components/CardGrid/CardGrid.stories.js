import React from 'react';
import { action } from '@storybook/addon-actions';
import CardGrid from '.';

export const byDefault = () => <CardGrid />;

export default {
  title: 'CardGrid',
  component: CardGrid,
};
