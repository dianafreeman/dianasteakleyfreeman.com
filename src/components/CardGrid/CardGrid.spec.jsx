import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react'
import CardGrid from '.';

describe('CardGrid', () => {

  it('should render without exploding', () => {
    expect(render(<CardGrid />)).not.to.throw
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<CardGrid />).toJSON()
    jestExpect(tree).toMatchSnapshot();
  });
});