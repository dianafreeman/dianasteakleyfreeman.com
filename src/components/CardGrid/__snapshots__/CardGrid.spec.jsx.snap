// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGrid should match snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-height: 100px;
  margin: 10px;
}

.c1 {
  width: 50%;
  position: relative;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
  background-color: #B7B7B7;
  min-width: 33vw;
}

.c2 {
  color: #ffffff;
  width: 100%;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  position: absolute;
  bottom: 0;
  margin-bottom: 0.25rem;
}

<div
  onClick={[Function]}
>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <h3
        className="c2"
      >
        item a
      </h3>
    </div>
  </div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <h3
        className="c2"
      >
        item b
      </h3>
    </div>
  </div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <h3
        className="c2"
      >
        item c
      </h3>
    </div>
  </div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <h3
        className="c2"
      >
        item d
      </h3>
    </div>
  </div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <h3
        className="c2"
      >
        item e
      </h3>
    </div>
  </div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <h3
        className="c2"
      >
        item f
      </h3>
    </div>
  </div>
</div>
`;
